include/tvm/relay/pass.h

```diff
@@ -156,7 +175,74 @@ size_t StructuralHash(const Type& type);
  */
 size_t StructuralHash(const Expr& expr);
 
+/*! \brief The hash struct for expressions. */
+struct ExprHash {
```

Comment:
can we just expose StructuralHash struct and expose that interface?

Reply:
It makes overloading more annoying, because we would need to introduce separate structures or template to deal with Expr and Type.